core/api/wallet/ledger.cpp

```diff
@@ -22,8 +22,34 @@ namespace fc::api {
 
   Ledger::Ledger(const MapPtr &store) : store(store) {}
 
-  bool Ledger::Has(const Address &address) const {
-    return store->contains(encode(address));
+  outcome::result<bool> Ledger::Has(const Address &address) const {
+    if (store->contains(encode(address))) {
+      OUTCOME_TRY(value, store->get(encode(address)));
+      OUTCOME_TRY(j_file, codec::json::parse(value));
+      OUTCOME_TRY(ledger_key_info, decode<LedgerKeyInfo>(j_file));
+
+      auto [app, err] = LedgerFilecoinManager::FindLedgerFilecoinApp();
+      if (err != std::nullopt) {
+        return app_error;
+      }
+
+      std::string addr;
+      std::tie(std::ignore, std::ignore, addr, err) =
+          app->GetAddressPubKeySECP256K1(ledger_key_info.path);
+      if (err != std::nullopt) {
+        OUTCOME_TRY(store->remove(encode(address)));
```

Comment:
may be dangerous (eg user who has two ledgers and inserted wrong one will lose keys from other device).  

don't remove.  
there is explicit api `WalletDelete` for removal
```suggestion
```